main.cpp

```diff
@@ -4,36 +4,63 @@
 #include "Devices/acunit.h"
 #include "smarthomestatus.h"
 
-#include "uniqueptr.h"
 #include "andreea_unique_ptr"
 
 int main()
 {
     
     smart_home::SmartHome smartHome;
-    for (auto i = 0; i < 6; i++)
-    {
+
         try
         {
-            std::string filename = "../data_files/data" + std::to_string(i) + ".xml";
+            std::string filename = "../data_files/data0.xml";
             std::cout<<filename<<std::endl<<std::endl;
             smartHome.GetDataFromXml( filename.c_str());
             
-            smart_home::MyUniquePtr<smart_home::Room> room = smartHome.GetRoom("LivingRoom").value();
 
-            const std::vector<smart_home::StatusObject*> sensors = room->GetSensors();
-            const std::vector<smart_home::StatusObject*> devices = room->GetDevices();
+            // const std::vector<smart_home::StatusObject*> sensors = room->GetSensors();
+            // const std::vector<smart_home::StatusObject*> devices = room->GetDevices();
+
+            // smart_home::SmartHomeStatus status;
+
+            // std::thread sensorsThread([&sensors, &status]() {
+            //     status.PrintStatus(sensors, "Sensors");
+            //     });
+
+            // std::thread devicesThread([&devices, &status]() {
+            //     status.PrintStatus(devices, "Devices");
+            //     });
+
+            // sensorsThread.detach();
+            // devicesThread.detach();
+
+            smart_home::SmartHomeStatus status;
+            
+            std::thread statusThread([&smartHome, &status]() {
+                    status.PrintSmartHomeStatus(smartHome);
+
+                });
+
+            statusThread.detach();
```

Comment:
try to use the threads with join, not detach as you share data between them.
Also there is no synchronisation need between the threads (even though you sleep the main thread and do changes). Look up lock_guard, unique_lock, read about mutexes and how you should include them in your project to make use of them

Reply:
I tried to use that thread with join, but if I do that my main thread sleeps, and I understood from my homework that I should make changes without any issues, so how am I supposed to do that if my main thread sleeps when I do join?
I used unique_lock and mutex in the SmartHomeStatus class, and I understood that they are used for avoiding data races, but in this case I don't know how they can be useful 